src/main/java/edu/kit/compiler/transform/IRStatementVisitor.java

```diff
@@ -0,0 +1,187 @@
+        Node getCond = con.newCond(cmp);
+        Node projTrue = con.newProj(getCond, Mode.getX(), 1);
+        Node projFalse = con.newProj(getCond, Mode.getX(), 0);
+
+        Block finalBlock = con.newBlock();
+
+        boolean hasElse = ifStmt.getElseStatement().isPresent();
+        if (hasElse) {
+            // else block
+            Block elseBlock = con.newBlock();
+            elseBlock.addPred(projFalse);
+            elseBlock.mature();
+            con.setCurrentBlock(elseBlock);
+            if (!ifStmt.getElseStatement().get().accept(this)) {
+                // block does not return
+                Node elseJmp = con.newJmp();
+                finalBlock.addPred(elseJmp);
+            }
+        } else {
+            finalBlock.addPred(projFalse);
+        }
+
+        // then block
+        Block thenBlock = con.newBlock();
+        thenBlock.addPred(projTrue);
+        thenBlock.mature();
+        con.setCurrentBlock(thenBlock);
+        if (!ifStmt.getThenStatement().accept(this)) {
+            // block does not return
+            Node thenJmp = con.newJmp();
+            finalBlock.addPred(thenJmp);
+        }
+
+        // final block
+        finalBlock.mature();
+        con.setCurrentBlock(finalBlock);
+
+        // TODO: can this cause problems?
+        return false;
+    }
+
+    @Override
+    public Boolean visit(WhileStatementNode whileStmt) {
+        Construction con = context.getConstruction();
+        Node jmp = con.newJmp();
+        Block loopHeader = con.newBlock();
+        loopHeader.addPred(jmp);
+        con.setCurrentBlock(loopHeader);
+
+        // endless loop: ensure construction of mem node and keep block
+        con.getCurrentMem();
+        con.getGraph().keepAlive(loopHeader);
+
+        // TODO: proper condition evaluation
+        ExpressionNode cond = whileStmt.getCondition();
+        Node left = evalExpression(cond);
+        Node right = con.newConst(0, getMode(cond.getResultType()));
+        Node cmp = con.newCmp(left, right, Relation.LessGreater);
+        Node getCond = con.newCond(cmp);
+        Node projTrue = con.newProj(getCond, Mode.getX(), 1);
+        Node projFalse = con.newProj(getCond, Mode.getX(), 0);
+
+        Block loopBody = con.newBlock();
+        loopBody.addPred(projTrue);
+        loopBody.mature();
+        con.setCurrentBlock(loopBody);
+        if (!whileStmt.getStatement().accept(this)) {
+            // block does not return
+            Node backJmp = con.newJmp();
+            loopHeader.addPred(backJmp);
+        }
+        loopHeader.mature();
+
+        Block nextBlock = con.newBlock();
+        nextBlock.addPred(projFalse);
+        nextBlock.mature();
+        con.setCurrentBlock(nextBlock);
+        return false;
+    }
+
+    private Node evalExpression(ExpressionNode expr) {
+        Construction con = context.getConstruction();
+        if (expr instanceof ExpressionNode.IdentifierExpressionNode) {
```

Comment:
Should this really be part of the statement visitor, or do you plan on removing it, once  you have integrated the expression visitor?

Reply:
This will be removed with the expression visitor. However, until then it is required for the unit tests to work

Reply:
Well, I could instead let the tests fail, but I'm not sure whether that is better

Reply:
In that case I think it's fine as is. Just noticed `instanceof` and wanted to make sure that wasn't permanent.